controllers/tasks.js

```diff
@@ -3,29 +3,91 @@ const Task = require('../models/Tasks')
 const {BadRequestError, NotFoundError} = require('../errors');
 
 const getAllTasks = async (req, res) => {
-    res.send('get all tasks');
+    const tasks = await Task.find({createdBy: req.user.userId}).sort('createdAt');
+    res.status(StatusCodes.OK).json({success:true, data:{tasks, count: tasks.length }});
 }
+
+const getListTasks = async (req, res) => {
+    const listId = req.params.listId; 
+    const tasks = await Task.find({
+        createdBy: req.user.userId, 
+        listIds: listId
+    }).sort('createdAt');
+
+    res.status(StatusCodes.OK).json({success:true, data:{tasks, count: tasks.length }});
+}
+
 const getTask = async (req, res) => {
-    res.send('get task');
+    const {user:{userId}, params: {id: taskId}} = req; 
+    const task = await Task.find({
```

Comment:
Need to use findOne here.  Otherwise, task is assign an empty list and the NotFoundError is not thrown